pkg/storage/test/tuples.go

```diff
@@ -317,6 +318,98 @@ func TupleWritingAndReadingTest(t *testing.T, datastore storage.OpenFGADatastore
 		_, err = gotTuples.Next()
 		require.ErrorIs(t, err, storage.ErrIteratorDone)
 	})
+
+	t.Run("reading_userset_tuples_with_filter_made_of_direct_relation_reference", func(t *testing.T) {
+		storeID := ulid.Make().String()
+		tks := []*openfgapb.TupleKey{
+			tuple.NewTupleKey("document:1", "viewer", "user:*"),
+			tuple.NewTupleKey("document:1", "viewer", "users:*"),
+			tuple.NewTupleKey("document:1", "viewer", "group:eng#member"),
+			tuple.NewTupleKey("document:1", "viewer", "grouping:eng#member"),
+		}
+
+		err := datastore.Write(ctx, storeID, nil, tks)
+		require.NoError(t, err)
+
+		gotTuples, err := datastore.ReadUsersetTuples(ctx, storeID, storage.ReadUsersetTuplesFilter{
+			Object:   "document:1",
+			Relation: "viewer",
+			AllowedUserTypeRestrictions: []*openfgapb.RelationReference{
+				typesystem.DirectRelationReference("group", "member"),
+			},
+		})
+		require.NoError(t, err)
+
+		iter := storage.NewTupleKeyIteratorFromTupleIterator(gotTuples)
+		defer iter.Stop()
+
+		var gotTupleKeys []*openfgapb.TupleKey
+		for {
+			tk, err := iter.Next()
```

Comment:
I suggest simplifying this to

```
tk, err := iter.Next()
require.NoError(t, err)
require.Equal(t, tk, tuple.NewTupleKey("document:1", "viewer", "group:eng#member")

_, err := iter.Next()
require.ErrorIs(t, err, storage.ErrIteratorDone)
```
This is because you expect the iterator to only yield one value and then the second time the iterator should yield `storage.ErrIteratorDone`.